### PR TITLE
Moving top level files around.

### PR DESCRIPTION
**Pull request by kberg**
To start to make better tsconfig.jsons, I'm going to move many of the things that are source out of the top directory.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -41,16 +41,17 @@
...
     "compile": "tsc",
     "cover": "nyc mocha --file build/tests/utils/Vue.js --recursive build/tests",
-    "fix": "eslint --fix src server.ts script.ts tests --ext ts",
-    "lint": "eslint src server.ts script.ts tests --ext ts",
+    "fix": "eslint --fix src tests --ext ts",
+    "lint": "eslint src tests --ext ts",
+    "makeJson": "node make_static_json.js",
     "precover": "tsc --build tsconfig-test.json",
-    "prebuild": "node make_static_json.js && lessc src/styles/common.less styles.css",
+    "prebuild": "mkdir -p build/ && lessc src/styles/common.less > build/styles.css",
```
[the remark this answers is not in the document]
Comment (bafolts) [reply package.json:52]: I haven't used windows in a while. Not sure if `mkdir -p` will be a problem on windows machines.

**Review by bafolts — COMMENTED** [review]
I don't know if this is going to break build for windows users. Let me see if I can add a github action to build on a windows machine as well so we always consider windows users in some way.

**Review by bafolts — CHANGES_REQUESTED** [review]
Can you make a small change to add windows machine to our build? I don't know if this change will break windows and with more things moving around we will be putting windows at more risk. The change was easy.

Change:
https://github.com/bafolts/terraforming-mars/compare/use-windows-build

Sample Run:
https://github.com/bafolts/terraforming-mars/runs/1738975654

**Comment by kberg** [comment]
In conversation with @bafolts I've merged #2501 into this one so as to resolve both at once.

**Review by bafolts — APPROVED** [review]
with added windows build step my concern is gone.

**Pull request merged**